packages/amplify-e2e-core/src/categories/api.ts

```diff
@@ -298,3 +298,244 @@ export function addRestApi(cwd: string, settings: any) {
+      .sendCarriageReturn()
+      .wait('Provide API name:')
+      .sendCarriageReturn()
+      .wait('Choose the default authorization type for the API')
+      .send(KEY_UP_ARROW)
+      .send(KEY_UP_ARROW)
+      .sendCarriageReturn()
+      .wait('Do you want to configure advanced settings for the GraphQL API')
+      .sendCarriageReturn()
+      .wait('Do you have an annotated GraphQL schema?')
+      .sendLine('n')
+      .wait('Do you want a guided schema creation')
+      .sendLine('y')
+      .wait('What best describes your project')
+      .sendCarriageReturn()
+      .wait('Do you want to edit the schema now?')
+      .sendLine('n')
+      .wait('"amplify publish" will build all your local backend and frontend resources')
+      .run((err: Error) => {
+        if (!err) {
+          resolve();
+        } else {
+          reject(err);
+        }
+      });
+  });
+}
+
+export function addApiWithOIDCAuthType(
+  projectDir: string,
+  oidcProviderName: string,
+  oidcProviderDomain: string,
+  oidcClientId: string,
+  ttlaIssueInMillisecond: string,
+  ttlaAuthInMillisecond: string,
+) {
+  return new Promise((resolve, reject) => {
+    spawn(getCLIPath(), ['add', 'api'], { cwd: projectDir, stripColors: true })
+      .wait('Please select from one of the below mentioned services:')
+      .sendCarriageReturn()
+      .wait('Provide API name:')
+      .sendCarriageReturn()
+      .wait('Choose the default authorization type for the API')
+      .send(KEY_DOWN_ARROW)
+      .send(KEY_DOWN_ARROW)
+      .send(KEY_DOWN_ARROW)
+      .sendCarriageReturn()
+      .wait('Enter a name for the OpenID Connect provider')
+      .send(oidcProviderName)
+      .sendCarriageReturn()
+      .wait('Enter the OpenID Connect provider domain (Issuer URL)')
+      .send(oidcProviderDomain)
+      .sendCarriageReturn()
+      .wait('Enter the Client Id from your OpenID Client Connect application (optional)')
+      .send(oidcClientId)
+      .sendCarriageReturn()
+      .wait('Enter the number of milliseconds a token is valid after being issued to a user')
+      .send(ttlaIssueInMillisecond)
+      .sendCarriageReturn()
+      .wait('Enter the number of milliseconds a token is valid after being authenticated')
+      .send(ttlaAuthInMillisecond)
+      .sendCarriageReturn()
+      .wait('Do you want to configure advanced settings for the GraphQL API')
+      .sendCarriageReturn()
+      .wait('Do you have an annotated GraphQL schema?')
+      .sendLine('n')
+      .wait('Do you want a guided schema creation')
+      .sendLine('y')
+      .wait('What best describes your project')
+      .sendCarriageReturn()
+      .wait('Do you want to edit the schema now?')
+      .sendLine('n')
+      .wait('"amplify publish" will build all your local backend and frontend resources')
+      .run((err: Error) => {
+        if (!err) {
+          resolve();
+        } else {
+          reject(err);
+        }
+      });
+  });
+}
```

Comment:
Can we consolidate all of this into a single function which accepts an auth configuratation and then select the appropriate auth type without having 3 separate function which has so much code repeated

Reply:
done